src/main/java/bio/terra/workspace/app/controller/GoogleReferenceController.java

```diff
@@ -0,0 +1,245 @@
+package bio.terra.workspace.app.controller;
+
+import bio.terra.workspace.common.utils.ControllerTranslationUtils;
+import bio.terra.workspace.generated.controller.GoogleReferenceApi;
+import bio.terra.workspace.generated.model.BigQueryDatasetReference;
+import bio.terra.workspace.generated.model.CreateBigQueryDatasetReferenceRequestBody;
+import bio.terra.workspace.generated.model.CreateGoogleBucketReferenceRequestBody;
+import bio.terra.workspace.generated.model.DataReferenceRequestMetadata;
+import bio.terra.workspace.generated.model.GoogleBucketReference;
+import bio.terra.workspace.service.datareference.DataReferenceService;
+import bio.terra.workspace.service.datareference.model.CloningInstructions;
+import bio.terra.workspace.service.datareference.model.DataReference;
+import bio.terra.workspace.service.datareference.model.DataReferenceRequest;
+import bio.terra.workspace.service.datareference.model.DataReferenceType;
+import bio.terra.workspace.service.datareference.model.ReferenceObject;
+import bio.terra.workspace.service.datareference.utils.DataReferenceValidationUtils;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequestFactory;
+import java.util.UUID;
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+
+@Controller
+public class GoogleReferenceController implements GoogleReferenceApi {
+
+  private final DataReferenceValidationUtils dataReferenceValidation;
+  private final DataReferenceService dataReferenceService;
+  private final AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+  private final HttpServletRequest request;
+
+  @Autowired
+  public GoogleReferenceController(
+      DataReferenceValidationUtils dataReferenceValidation,
+      DataReferenceService dataReferenceService,
+      AuthenticatedUserRequestFactory authenticatedUserRequestFactory,
+      HttpServletRequest request) {
+    this.dataReferenceValidation = dataReferenceValidation;
+    this.dataReferenceService = dataReferenceService;
+    this.authenticatedUserRequestFactory = authenticatedUserRequestFactory;
+    this.request = request;
+  }
+
+  private final Logger logger = LoggerFactory.getLogger(GoogleReferenceController.class);
+
+  private AuthenticatedUserRequest getAuthenticatedInfo() {
+    return authenticatedUserRequestFactory.from(request);
+  }
```

Comment:
`DataRepoReferenceController` and `GoogleReferenceController` share boilerplate - is there opportunity for a `ReferenceController` base class?

Reply:
Done. I'm not sure I took the cleanest approach given the way I'm passing beans around, but it does cut down on boilerplate. I might try to revisit this a bit to shrink the argument list